### PR TITLE
Getting started templates

### LINKED ISSUE #235: Create skeleton projects for Java and Scala
**Is your feature request related to a problem? Please describe.**
Getting started with CloudFlow is frustrating because there's no easy way to get started with a working project you can build on. 

**Is your feature request related to a specific runtime of cloudflow or applicable for all runtimes?**
All runtimes 

**Describe the solution you'd like**
I'd like to be able to start from a simple Java or Scala skeleton project all configured with builds, Docker files, Avro schemas, etc. that I can then modify to my requirements. 

**Describe alternatives you've considered**
I'm trying to extract a working project from the examples, but it's not the easiest.

**Additional context**

### PR DESCRIPTION
**Pull request by maasg**
This PR adds easy-to-use templates with a minimal working implementation to serve as a starting point to develop applications with Cloudflow.
It adds a single-backend Scala and Java template.
(I'm adding the multi-backend as a separate issue)

TODO: Improve docs to point to the templates as a starting point.

@jasoncarreira Could you have a look to see if this is helpful?

Closes #235

### CODE REVIEW

**Comment by maasg** [comment]
CI build failed with
```
[error] /home/runner/work/cloudflow/cloudflow/examples/taxi-ride/logger/src/main/scala/taxiride/logger/FarePerRideLogger.scala:23:17: object flink is not a member of package taxiride
``` 
which is unrelated to this PR. Launching the build again.

File: examples/templates/single-backend-java/build.sbt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import sbt.Keys._
+
+lazy val templateJavaProject = (project in file("."))
+     //enable here the backend you want to use in your application:
+     //CloudflowAkkaStreamsApplicationPlugin, CloudflowSparkApplicationPlugin, CloudflowFlinkApplicationPlugin
+    .enablePlugins(CloudflowAkkaStreamsApplicationPlugin, ScalafmtPlugin)
+    .settings(
+      scalafmtOnCompile := true,
+      libraryDependencies ++= Seq(
+	      "ch.qos.logback"     %  "logback-classic"        % "1.2.3",
```
Comment (RayRoestenburg) [inline examples/templates/single-backend-java/build.sbt:11]: Some strange formatting

File: examples/templates/single-backend-java/src/main/java/com/example/app/ConsoleOutput.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  // check the Streamlet API of your chosen implementation to determine the entry point
+  // corresponding to your chosen backend.
+  public RunnableGraphStreamletLogic createLogic() {
+    return new RunnableGraphStreamletLogic(getContext()) {
+
+      public String format(Data data) {
+        return data.id() + "-> " +data.value();
+      }
+      public RunnableGraph<NotUsed> createRunnableGraph() {
+        return getPlainSource(inlet)
```
Comment (RayRoestenburg) [inline examples/templates/single-backend-java/src/main/java/com/example/app/ConsoleOutput.java:55]: Maybe use resetposition earliest

File: examples/templates/single-backend-scala/build.sbt [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import sbt._
+import sbt.Keys._
+
+lazy val templateScala = (project in file("."))
+    .enablePlugins(CloudflowAkkaStreamsApplicationPlugin, ScalafmtPlugin)
+    .settings(
+      scalafmtOnCompile := true,
+      libraryDependencies ++= Seq(
+	      "ch.qos.logback"     %  "logback-classic"        % "1.2.3",
+        "org.scalatest"      %% "scalatest"              % "3.0.8" % "test"
```
Comment (RayRoestenburg) [inline examples/templates/single-backend-scala/build.sbt:10]: Formatting minor

File: examples/templates/single-backend-java/build.sbt (hunk 1 above)
Comment (maasg) [reply examples/templates/single-backend-java/build.sbt:11]: weird. It looks OK on my editor. 
image

File: examples/templates/single-backend-scala/build.sbt (hunk 3 above)
Comment (maasg) [reply examples/templates/single-backend-scala/build.sbt:10]: Same as previous 
image

File: examples/templates/single-backend-java/src/main/java/com/example/app/ConsoleOutput.java (hunk 2 above)
Comment (maasg) [reply examples/templates/single-backend-java/src/main/java/com/example/app/ConsoleOutput.java:55]: Is that needed in a template?  Hopefully, all the code here will be replaced by user code.

**Pull request merged**